Mathlib/Data/Sym/Basic.lean

```diff
@@ -532,6 +532,45 @@ theorem mem_append_iff {s' : Sym α m} : a ∈ s.append s' ↔ a ∈ s ∨ a ∈
   Multiset.mem_add
 #align sym.mem_append_iff Sym.mem_append_iff
 
+/-- An element of type α can be seen as an element of Sym α 1 --/
+def one {α : Type*} (i : α) : Sym α 1 := ⟨[i], by simp only [Multiset.coe_singleton,
+  Multiset.card_singleton]⟩
+
+lemma one_injective : Function.Injective (Sym.one : α → Sym α 1) := by
+  intros i j h
+  have hi : (Sym.one i).1 = [i] := by
+      simp only [Sym.one, Multiset.coe_singleton]
+  have hj : (Sym.one j).1 = [j] := by
+    simp only [Sym.one, Multiset.coe_singleton]
+  rw [h] at hi
+  rw [hi] at hj
+  apply List.singleton_inj.mp
+  exact Multiset.coe_eq_singleton.mp hj
+
+lemma singleton_of_len_one (s : Sym α 1) : ∃! i, Sym.one i = s := by
+  have : ∃ i, s.toMultiset = {i} := by
+    simp_rw [Sym] at s
+    apply Multiset.card_eq_one.mp
+    exact s.2
+  obtain ⟨i, hi⟩ := this
+  have : s = one i := by
+    simp only [Sym.one, Multiset.coe_singleton]
+    simp_rw [← hi]
+    rfl
+  use i
+  constructor
+  · exact this.symm
+  · intro j hj
+    apply Sym.one_injective
+    rw [hj]
+    exact this
+
+lemma one_bijective : Function.Bijective (Sym.one : α → Sym α 1) := by
+  rw [Function.bijective_iff_existsUnique]
+  exact Sym.singleton_of_len_one
+
+lemma one_equiv : α ≃ Sym α 1 := Equiv.ofBijective _ Sym.one_bijective
```

Comment:
Ah, I didn't notice you wanted this!
* First of all an `equiv` contains data (it's the fact that there is a bijection, it is *the* bijection), so it should be a (noncomputable) def.
You can maybe define directly as an equiv and forget about the bare function:
```lean
open Multiset Classical in
@[simps apply]
noncomputable def one_equiv : α ≃ Sym α 1 where
  toFun := fun i ↦ ⟨[i], by simp only [coe_singleton, card_singleton]⟩
  invFun := fun x ↦ choose <| card_eq_one.1 x.2
  left_inv := fun i ↦ by simpa using (choose_spec <| card_eq_one.1
      (⟨[i], by simp only [coe_singleton, card_singleton]⟩ : Sym α 1).2).symm
  right_inv := fun x ↦ by
    simp only [val_eq_coe, coe_singleton]
    exact ext <| by simp_rw [← choose_spec <| card_eq_one.1 x.2]; rfl
```

Reply:
It open a parenthesis that is autoclosed at the end of the command. So `f (g x y)` is the same as `f <| g x y`. At the beginning it makes thing more difficult to read, but when you get used to it is nice. There is also the less used variant `|>`.